app/common/commonMenu.js

```diff
@@ -77,20 +77,23 @@ module.exports.newPrivateTabMenuItem = () => {
     label: locale.translation('newPrivateTab'),
     accelerator: 'Shift+CmdOrCtrl+P',
     click: function (item, focusedWindow) {
-      // Check if Tor is available
-      const useTor = getSetting(settings.USE_TOR_PRIVATE_TABS)
-      if (useTor) {
-        ensureAtLeastOneWindow({
-          url: 'about:newtab',
-          isPrivate: true,
-          isTor: true
-        })
-      } else {
-        ensureAtLeastOneWindow({
-          url: 'about:newtab',
-          isPrivate: true
-        })
-      }
+      ensureAtLeastOneWindow({
+        url: 'about:newtab',
+        isPrivate: true
+      })
+    }
+  }
+}
+
+module.exports.newTorTabMenuItem = () => {
+  return {
+    label: locale.translation('newTorTab'),
+    click: function (item, focusedWindow) {
+      ensureAtLeastOneWindow({
+        url: 'about:newtab',
+        isPrivate: true,
+        isTor: true
```

Comment:
This will not result in a tor tab when there are 0 windows open, which is a common macOS scenario (as judging from the number of issues that were raised when I had a similar bug!).

In order to fix this we simply need to make the `openFramesInWindow` function inside _app/browser/windows.js_ aware of the `tor: ` property for the `createTabRequested` options argument.

Reply:
good catch thanks!